clang-15 error: expected unqualified-id due to dirfd macro in macOS 14 SDK

Hi!

In file boost/process/detail/posix/handles.hpp:
```
boost/process/detail/posix/handles.hpp:36:20: error: expected unqualified-id
    auto my_fd = ::dirfd(dir.get());
                   ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/dirent.h:136:21: note: expanded from macro 'dirfd'
#define dirfd(dirp) ({                         \
                    ^
In file included from [...]:
In file included from boost/process/child.hpp:25:
In file included from boost/process/posix.hpp:8:
In file included from boost/process/detail/posix/fd.hpp:15:
In file included from boost/process/detail/used_handles.hpp:14:
boost/process/detail/posix/handles.hpp:36:20: error: unexpected type name 'DIR': expected expression
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/dirent.h:137:5: note: expanded from macro 'dirfd'
    DIR *_dirp = (dirp);                       \
    ^
In file included from [...]:
In file included from boost/process/child.hpp:25:
In file included from boost/process/posix.hpp:8:
In file included from boost/process/detail/posix/fd.hpp:15:
In file included from boost/process/detail/used_handles.hpp:14:
boost/process/detail/posix/handles.hpp:36:20: error: use of undeclared identifier '_dirp'
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/dirent.h:137:10: note: expanded from macro 'dirfd'
    DIR *_dirp = (dirp);                       \
         ^
In file included from [...]:
In file included from boost/process/child.hpp:25:
In file included from boost/process/posix.hpp:8:
In file included from boost/process/detail/posix/fd.hpp:15:
In file included from boost/process/detail/used_handles.hpp:14:
boost/process/detail/posix/handles.hpp:120:24: error: expected unqualified-id
        auto my_fd = ::dirfd(dir);
                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/dirent.h:136:21: note: expanded from macro 'dirfd'
#define dirfd(dirp) ({                         \
                    ^
In file included from [...]:
In file included from boost/process/child.hpp:25:
In file included from boost/process/posix.hpp:8:
In file included from boost/process/detail/posix/fd.hpp:15:
In file included from boost/process/detail/used_handles.hpp:14:
boost/process/detail/posix/handles.hpp:120:24: error: unexpected type name 'DIR': expected expression
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/dirent.h:137:5: note: expanded from macro 'dirfd'
    DIR *_dirp = (dirp);                       \
    ^
In file included from [...]:
In file included from boost/process/child.hpp:25:
In file included from boost/process/posix.hpp:8:
In file included from boost/process/detail/posix/fd.hpp:15:
In file included from boost/process/detail/used_handles.hpp:14:
boost/process/detail/posix/handles.hpp:120:24: error: use of undeclared identifier '_dirp'

/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/dirent.h:137:10: note: expanded from macro 'dirfd'
    DIR *_dirp = (dirp);                       \
         ^
6 errors generated.
```

--- a/include/boost/process/detail/posix/handles.hpp
+++ b/include/boost/process/detail/posix/handles.hpp
@@ -33,7 +33,7 @@ inline std::vector<native_handle_type> get_handles(std::error_code & ec)
     else
         ec.clear();
 
-    auto my_fd = ::dirfd(dir.get());
+    auto my_fd = dirfd(dir.get());
 
     struct ::dirent * ent_p;
 
@@ -117,7 +117,7 @@ struct limit_handles_ : handler_base_ext
             return;
         }
 
-        auto my_fd = ::dirfd(dir);
+        auto my_fd = dirfd(dir);
         struct ::dirent * ent_p;
 
         while ((ent_p = readdir(dir)) != nullptr)

